File: 2025-lua/01.lua
local time_start = os.clock()
local p = 50
local pt1 = 0
local pt2 = 0

for line in io.lines() do
    local dir = string.sub(line, 0, 1)
    local amount = string.sub(line, 2)

    if dir  == "L" then
        p = p - amount

        while p < 0 do
            p = 100 + p
            pt2 = pt2 + 1
        end
    else
        p = p + amount
        while p > 99 do
            p = 0 + (p - 100)
            pt2 = pt2 + 1
        end
    end

    if p == 0 then
        pt1 = pt1 + 1
    end

end

print("--- Day 1: Secret Entrance ---")
print("Part 1", pt1)
print("Part 2", pt2)
print("Took ", os.clock() - time_start, "ms")

File: 2025-lua/01.py
import sys
import time
import math

time_start = time.time_ns()
pt1 = 0
pt2 = 0
p = 50

lines = [l.strip() for l in sys.stdin.readlines()]

for l in lines:
	amount = -int(l[1:]) if l[0] == 'L' else int(l[1:])
	p = p + amount
	pt2 = pt2 + abs(p // 100)
	p = p % 100
	if p == 0:
		pt1 += 1


print("--- Day 1: Secret Entrance ---")
print("Part 1", pt1)
print("Part 2", pt2)
print("Took", (time.time_ns() - time_start) // 1_000_000, "ms")

assert(pt1 == 999)
assert(pt2 == 6099)
